From c958f2eb95a144ecbfcf3b7f18614fab553a0186 Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Sun, 3 Oct 2021 23:21:09 +0900
Subject: [PATCH] get_exo_preferred_file_manager: check if the requested key
 exists before actually use it

In get_exo_preferred_file_manager(), when ~/.config/xfce4/helpers.rc exists,
this function gets application settings from this file, then tries to use
the value for "FileManager" key.

However in some cases (for example, using `$ xfce4-mime-settings`, changing
"Terminal Emulator" only), this rc file does not contain "FileManager" key.
Currently in this case, catfish gets traceback.

To fix this, check if "FileManager" key exists before actually using it.

Closes #64 .
---
 catfish/CatfishWindow.py | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/catfish/CatfishWindow.py b/catfish/CatfishWindow.py
index c199353..3d855d0 100644
--- a/catfish/CatfishWindow.py
+++ b/catfish/CatfishWindow.py
@@ -1213,16 +1213,16 @@ class CatfishWindow(Window):
             cfg = "%s/xfce4/helpers.rc" % config_dir
             if os.path.exists(cfg):
                 apps = self.get_exo_preferred_applications(cfg)
-                if 'custom-FileManager' in apps['FileManager']:
-                    with open(custFM) as f:
-                        for line in f:
-                            CFM = line.replace('X-XFCE-Commands=', '').strip()
-                            if 'X-XFCE-Commands=' in line:
-                                return CFM
-
                 if 'FileManager' in apps:
+                    if 'custom-FileManager' in apps['FileManager']:
+                        with open(custFM) as f:
+                            for line in f:
+                                CFM = line.replace('X-XFCE-Commands=', '').strip()
+                                if 'X-XFCE-Commands=' in line:
+                                    return CFM
                     return apps['FileManager']
 
+
         return "Thunar"
 
     def using_thunar_fm(self):
-- 
2.32.0

